### ISSUE TITLE
fatal error: 'gtk/gtk.h' file not found

### ISSUE THREAD

**Issue by mrmodolo** [wsdjeg/SpaceVim#1162]
## Autocomplete with gtk files

Example:
#include<gtk/gtk.h>

int main(int argc, char **argv) {
  GtkWidget *win;
  gtk_init(&argc, &argv);
  win = gtk_window_new(GTK_WINDOW_TOPLEVEL);
  gtk_window_set_title(GTK_WINDOW(win), "Hello there");
  g_signal_connect(win, "destroy", G_CALLBACK(gtk_main_quit), NULL);
  gtk_widget_show(win);
  gtk_main();

}

I created the file '.clang_complete' and still continue with the error gtk/gtk.h not found!

Below the '.clang_complete' file contents:
-pthread
-I/usr/include/gtk-3.0
-I/usr/include/at-spi2-atk/2.0
-I/usr/include/at-spi-2.0
-I/usr/include/dbus-1.0
-I/usr/lib/x86_64-linux-gnu/dbus-1.0/include
-I/usr/include/gtk-3.0
-I/usr/include/gio-unix-2.0/
-I/usr/include/mirclient
-I/usr/include/mircore
-I/usr/include/mircookie
-I/usr/include/cairo
-I/usr/include/pango-1.0
-I/usr/include/harfbuzz
-I/usr/include/pango-1.0
-I/usr/include/atk-1.0
-I/usr/include/cairo
-I/usr/include/pixman-1
-I/usr/include/freetype2
-I/usr/include/libpng12
-I/usr/include/gdk-pixbuf-2.0
-I/usr/include/libpng12
-I/usr/include/glib-2.0
-I/usr/lib/x86_64-linux-gnu/glib-2.0/include
-lgtk-3
-lgdk-3
-lpangocairo-1.0
-lpango-1.0
-latk-1.0
-lcairo-gobject
-lcairo
-lgdk_pixbuf-2.0
-lgio-2.0
-lgobject-2.0
-lglib-2.0

## Environment Information
- OS: Distributor ID: Ubuntu
Description:    Ubuntu 16.04.3 LTS
Release:        16.04
Codename:       xenial

- vim version: VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Nov 24 2016 16:44:48)
![captura de tela de 2017-12-22 14-19-59](https://user-images.githubusercontent.com/1923174/34304975-40cbdf34-e723-11e7-8625-9d4801704cf2.png)

**Comment by wsdjeg** [comment]
you need read https://spacevim.org/layers/lang/c/

**Comment by mrmodolo** [comment]
[hello_gtk.zip](https://github.com/SpaceVim/SpaceVim/files/1584350/hello_gtk.zip)
Hi!
I read!
But I still have the same error!
I created a .SpaceVim.d/init.vim file with the commands below:
call SpaceVim#layers#load('lang#c', {'enable_libclang' : 1,'libclang_path' : '/usr/lib/libclang.so',})
I also created the .clang and compile_flags.txt files!
I'm attaching the files, maybe you can help me!

Thank you and Merry Christmas.

**Comment by wsdjeg** [comment]
I will check it later, and Thanks, Merry Christmas to you too.

**Comment by wsdjeg** [comment]
@mrmodolo got it, because some plugin are lazzy loaded on `InsertEnter` event,  so switch to Insert mode, them go back to normal mode, save file, I think the error wil gone.

**Comment by wsdjeg** [comment]
ok, I will fix it. It seems a bug.

**Comment by mrmodolo** [comment]
Thank you very much!

**Comment by wsdjeg** [comment]
same errors: https://github.com/tweekmonster/deoplete-clang2/issues/5

**Comment by wsdjeg** [comment]
Fixed in #1169

**Comment by mrmodolo** [comment]
Hi!

I still keep getting the error!
fatal error: 'gtk/gtk.h' file no fount
![image](https://user-images.githubusercontent.com/1923174/34327814-81f5e322-e8b5-11e7-93c8-a747a7a29bda.png)

I also noticed that autocomplete only shows up to /usr/include/gtk-3.0!

![image](https://user-images.githubusercontent.com/1923174/34327822-abe44728-e8b5-11e7-8bb7-069a37c017a4.png)

Thanks,
Marcelo Módolo

**Comment by mrmodolo** [comment]
Hi!

I forgot to say that before I ran the command ':SPUpdate'.

Thanks

**Comment by mrmodolo** [comment]
Hi!

I also tested with neovim (after pip3 install neovim) and the behavior was the same!
I tested it on two different computers, one with Manjaro XFCE and another with Manjaro Gnome.

I created the '.ctags' file with the flags obtained by the command output:

`$ pkg-config --cflags --libs gtk+-3.0 | tr ' ' '\n`

You can test by creating a `*.c` file with the include below:

`#include <gtk/gtk.h>`

Immediately after saving the file with ':w' the error message will appear!

Thanks,
Marcelo Módolo

**Comment by wsdjeg** [comment]
@mrmodolo I think you need to update spacevim,  you should make sure .clang is in the root of your project.

**Comment by wsdjeg** [comment]
https://asciinema.org/a/sxWItf8xtuE2JCZDNm1zXA0ub

**Comment by mrmodolo** [comment]
Hi!

Have you tried using autocomplete after creating the ".clang" file?
I say this because even with an empty ".clang" file the error does not appear anymore, but I do not have autocomplete anymore!
If you can, give me the contents of the .clang file you created and please try the autocomplete feature.

Thank you!

**Comment by wsdjeg** [comment]
can you please use new issue for autocompletion? The title of this issue is failed to find *h file. :smile:

**Comment by mrmodolo** [comment]
Hi!

Yes, I can, without any problem! But I believe it's the same mistake. The .clang file seems to prevent the plugin from working and with this, the gtk.h error not found is no longer displayed!

I tried to reinstall SpaceVim but something happened and the installation seems to be in trouble!

The installation happens but after starting the vim and after updating all the plugins I no longer have the same splash screen and if I try the SPUpdate command several errors are presented.

Before trying to install I did the procedures for removal!

```
modolo@gsat019046:~$ curl -sLf https://spacevim.org/install.sh | bash -s -- --uninstall
modolo@gsat019046:~$ rm -rf .vim*
modolo@gsat019046:~$ rm -rf .SpaceVim*
modolo@gsat019046:~$ rm -rf .cache/vimfile*
```

Thanks,
Marcelo Módolo
